uservalidation/user_test.go

```diff
@@ -229,7 +219,12 @@ func TestCheckEmail(t *testing.T) {
 		},
 	}
 
-	executeTestCases(t, testCases)
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			resp := tt.testFunc(tt.input)
```

Comment:
Como `testFunc` não varia, dá pra chamar `CheckEmail` direto aqui e economizar umas linhas de código.

```suggestion
			resp := CheckEmail(tt.input)
```